parsing/procq.ml

```diff
@@ -256,7 +256,7 @@ let eoi_entry en =
   let symbs = Rule.next (Rule.next Rule.stop (Symbol.nterm en)) (Symbol.token Tok.PEOI) in
   let act = fun _ x loc -> x in
   let ext = Fresh (Gramlib.Gramext.First, make_rule [Production.make symbs act]) in
-  safe_extend e ext;
+  grammar_extend e ext;
```

Comment:
How can this code work properly w.r.t. backtracking?

Reply:
It's a non backtracking grammar extension.
I considered replacing it with a `parse_eoi : 'a Entry.t -> parsable -> 'a` extending the state locally but cannot really be bothered for just tactician and Default Proof Using (only users AFAIK)